frontend/src/uxbox/main/data/workspace.cljs

```diff
@@ -1329,6 +1329,24 @@
             (rx/of (update-shape shape-id
                                  {:interactions []}))))))))
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; CANVAS OPTIONS
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defn change-canvas-color
+  [color]
+  (ptk/reify ::change-canvas-color
+    ptk/WatchEvent
+    (watch [_ state stream]
+      (rx/of (dwc/commit-changes
+              [{:type :set-option
+                :option :background
+                :value color}]
+              [{:type :set-option
+                :option :background
+                :value nil}]
```

Comment:
Shouldn't you remember the current color and put it in the undo command?

Reply:
You're right. Fixed